src/main/java/com/spotify/reaper/resources/ClusterResource.java

```diff
@@ -167,18 +169,39 @@ public Response addCluster(
     return viewCluster(newCluster.getName(), Optional.<Integer>absent(), Optional.of(createdURI));
   }
 
-  public Cluster createClusterWithSeedHost(String seedHost)
+  public Cluster createClusterWithSeedHost(String seedHostInput)
       throws ReaperException {
-    String clusterName;
-    String partitioner;
-    try (JmxProxy jmxProxy = context.jmxConnectionFactory.connect(seedHost)) {
-      clusterName = jmxProxy.getClusterName();
-      partitioner = jmxProxy.getPartitioner();
-    } catch (ReaperException e) {
-      LOG.error("failed to create cluster with seed host: {}", seedHost, e);
-      throw e;
+    Optional<String> clusterName = Optional.absent();
+    Optional<String> partitioner = Optional.absent();
+    Optional<List<String>> liveNodes = Optional.absent();
+    Set<String> seedHosts = parseSeedHosts(seedHostInput);
+    for(String seedHost:seedHosts) {
+      try (JmxProxy jmxProxy = context.jmxConnectionFactory.connect(seedHost)) {
+        clusterName = Optional.of(jmxProxy.getClusterName());
+        partitioner = Optional.of(jmxProxy.getPartitioner());
+        liveNodes = Optional.of(jmxProxy.getLiveNodes());
+        break;
+      } catch (ReaperException e) {
+        LOG.error("failed to create cluster with seed host: {}", seedHost, e);
+      }
     }
-    return new Cluster(clusterName, partitioner, Collections.singleton(seedHost));
+    
+    if(!clusterName.isPresent()) {
+      throw new ReaperException("Could not connect any seed host");
+    }
+    
+    
+    Set<String> seedHostsFinal = seedHosts;
+    if (context.config.getEnableDynamicSeedList() 
+        && liveNodes.isPresent()) {
+      seedHostsFinal = liveNodes.get().size()>0
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Use isEmpty() to check whether the collection is empty or not. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1155)